src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/projectcreation/model/ProjectCreationModel.kt

```diff
@@ -21,50 +22,81 @@ class ProjectCreationModel {
-    var selectedAnthology: Collection by property()
-    var selectedBook: Collection by property()
-    var anthologyList: ObservableList<Collection> by property(FXCollections.observableArrayList())
-    var bookList: ObservableList<Collection> by property(FXCollections.observableArrayList())
-
+    var collectionList: ObservableList<Collection> = FXCollections.observableArrayList()
     val languages: ObservableList<Language> = FXCollections.observableArrayList()
-    val resources: ObservableList<Collection> = FXCollections.observableArrayList()
+    var collectionStore: ArrayList<List<Collection>> = ArrayList()
+    var creationDepth: Int = 1
+    var bookDepthReached = false
 
     init {
         creationUseCase.getAllLanguages()
                 .observeOnFx()
                 .subscribe { retrieved ->
                     languages.setAll(retrieved)
                 }
+    }
 
+    fun getSourceRepos() {
         creationUseCase.getSourceRepos()
                 .observeOnFx()
                 .subscribe { retrieved ->
-                    resources.setAll(retrieved)
+                    collectionStore.add(retrieved.filter {
+                        it.resourceContainer?.language == sourceLanguageProperty
+                    })
+                    collectionList.setAll(collectionStore[collectionStore.size - 1])
                 }
     }
 
-    fun getResourceChildren() {
-        creationUseCase.getResourceChildren(selectedResource)
-                .observeOnFx()
-                .doOnSuccess {
-                    anthologyList.setAll(it.sortedBy { it.sort })
-                }
-                .subscribe()
+    fun checkLevel(selectedCollection: Collection, workspace: Workspace) {
+        if (collectionList[0].labelKey == "book") {
+            createProject(selectedCollection)
+            workspace.dock<ProjectHomeView>()
+        } else {
+            getResourceChildren(selectedCollection)
+        }
     }
 
-    fun getBooks() {
-        creationUseCase.getResourceChildren(selectedAnthology)
+
+    private fun getResourceChildren(parentCollection: Collection) {
+        creationUseCase.getResourceChildren(parentCollection)
                 .observeOnFx()
                 .doOnSuccess {
-                    bookList.setAll(it.sortedBy { it.sort })
+                    collectionStore.add(it)
+                    collectionList.setAll(collectionStore[collectionStore.size - 1].sortedBy { it.sort })
+                    getDepth(parentCollection)
                 }
                 .subscribe()
     }
 
-    fun createProject() {
+    fun getPreviousCollections(projectWizard: Wizard) {
+        if (collectionStore.size > 1) {
+            collectionStore.removeAt(collectionStore.size - 1)
+            collectionList.setAll(collectionStore[collectionStore.size - 1].sortedBy { it.sort })
+        } else {
+            projectWizard.back()
+        }
+    }
+
+    fun getDepth(selectedCollection: Collection) {
```

Comment:
Doesn't look like any critical uses of getDepth, creationDepth, or bookDepthReached. Can we get rid of all of them for simplicity?

Reply:
yeah I'll remove all that. I'm working on it still, i'll put it all back up once i'm done with it in a seperate PR